accepted/2020/net5/net5.md

```diff
@@ -0,0 +1,854 @@
+> Either target iOS 13.0, or use a version of the .NET SDK that supports iOS
+> 14.0. [BabyShark.csproj]
+
+## Requirements
+
+### Goals
+
+* Use naming that aligns with product strategy
+* Merge .NET Core and .NET Standard into a single concept
+* Developers should be able to understand compatibility relationships without
+  having to consult a mapping table.
+* Provide compatibility with existing concepts and NuGet packages
+* It would be great to get this into early previews of .NET 5
+* Support multi-targeting different versions of the same OS
+* Don't force multi-targeting for different versions of the same OS. It should
+  be possible to produce a single binary that can use newer APIs when calls are
+  properly guarded with an OS check.
+
+### Non-Goals
+
+* Replace TFMs or expand runtime identifiers (RIDs)
+
+## Design
+
+We'll have the following TFMs:
+
+| TFM             | Compatible With                                            | Comments                          |
+|-----------------|------------------------------------------------------------|-----------------------------------|
+| net5.0          | net1..4 (with NU1701 warning)                              | No WinForms or WPF                |
+|                 | netcoreapp1..3.1 (warning when WinForms/WPF is referenced) |                                   |
+|                 | netstandard1..2.1                                          |                                   |
+| net5.0-android  | xamarin.android                                            |                                   |
+|                 | (+everything else inherited from net5.0)                   |                                   |
+| net5.0-ios      | xamarin.ios                                                |                                   |
+|                 | (+everything else inherited from net5.0)                   |                                   |
+| net5.0-macos    | xamarin.mac                                                |                                   |
+|                 | (+everything else inherited from net5.0)                   |                                   |
+| net5.0-tvos     | xamarin.tvos                                               |                                   |
+|                 | (+everything else inherited from net5.0)                   |                                   |
+| net5.0-watchos  | xamarin.watchos                                            |                                   |
+|                 | (+everything else inherited from net5.0)                   |                                   |
+| net5.0-windows  | netcoreapp1..3.1                                           | WinForms + WPF                    |
+|                 | (+everything else inherited from net5.0)                   |                                   |
+| Tizen, Unity... | Will follow the Xamarin model                              |                                   |
```

Comment:
I'm wondering how Unity fits in here. Ideally, we'd like to provide an implementation that matches a proper subset of .NET 5. There will be some APIs we don't want to support. Can our implementation provide "hints" to Roslyn and IDEs about what APIs are not supported? Or should we provide a `net5.0-unity` TFM?

Note that I'm not talking about additional APIs that Unity exposes here (e.g. our `NativeArray` or  `NativeSlice`), but rather just about the BCL API.

Reply:
> I'm wondering how Unity fits in here. Ideally, we'd like to provide an implementation that matches a proper subset of .NET 5. There will be some APIs we don't want to support. Can our implementation provide "hints" to Roslyn and IDEs about what APIs are not supported? Or should we provide a `net5.0-unity` TFM?

By mid-next week, I'll have a spec for how we're planning on detecting platform issues like this from a Roslyn analyzer.

> Note that I'm not talking about additional APIs that Unity exposes here (e.g. our `NativeArray` or `NativeSlice`), but rather just about the BCL API.

Should we hop on a conf call next week to discuss how Unity might fit in? Based on your response to .NET Standard 2.1 I figured it's a bit too early to have that conversation, but given your interest it looks like that was a wrong assumption on my end.

Reply:
> Should we hop on a conf call next week to discuss how Unity might fit in? Based on your response to .NET Standard 2.1 I figured it's a bit too early to have that conversation, but given your interest it looks like that was a wrong assumption on my end.

Yes, I'd be happy to discuss this next week. We're not quite ready to take action yet, but I suspect things will be moving in the next few months, so I'm thinking about a plan now.

Reply:
@joshpeterson and me will chat tomorrow.